### DISCUSSION TITLE
Azure SSO: Error in handling response type

### DISCUSSION THREAD

**Discussion by janeerikengels** [OpenIDC/mod_auth_openidc#1035] [Q&A] [answered]
Hey there, we are trying for several weeks to get SSO against Azure done.
There must be something stupid inside what we have done ...
Any help for us?
![Screenshot 2023-03-22 121824](https://user-images.githubusercontent.com/93512164/226915282-56c099f2-084d-46c2-804f-feed7a8abe66.jpg)
[oidc_log_apache.txt](https://github.com/OpenIDC/mod_auth_openidc/files/11040262/oidc_log_apache.txt)
[httpd.conf.SSO.txt](https://github.com/OpenIDC/mod_auth_openidc/files/11040266/httpd.conf.SSO.txt)
Thank you !

**Comment by zandbelt** [comment, marked as answer]
your log has AD returning an error:

> AADSTS50146: This application is required to be configured with an application-specific signing key. It is either not configured with one, or the key has expired or is not yet valid."

and a Google search for AADSTS50146 returns https://stackoverflow.com/questions/59383452/aadsts50146-error-when-attempting-to-retrieve-oauth-access-token

**Reply by janeerikengels** [reply]
Sorry, that needed a long time ... we added the signing key with the help of the azure admins and had long discussions with microsoft as there returned claim did not contain UPN, but microsoft was telling that UPN is default and always there -> this is not the case -> we switched to authenticate the users via their email address
Thank you very much for your support!
